src/routes/session.ts

```diff
@@ -3,28 +3,70 @@ import {
 	findUser,
 	getUser,
 	comparePassword,
-	runInvalidPasswordComparison,
 } from '../platform/auth';
-import { sbvrUtils } from '../platform';
+import { resinApi, root, sbvrUtils } from '../platform';
 import { captureException, handleHttpErrors } from '../platform/errors';
 import { resetCounter } from '../lib/rate-limiting';
 import { RequestHandler } from 'express';
 import { SetupOptions } from '..';
 
 const { BadRequestError, NotFoundError } = sbvrUtils;
 
-export const whoami: RequestHandler = (req, res) =>
-	getUser(req)
-		.then(user =>
-			res.send({ id: user.id, username: user.username, email: user.email }),
-		)
-		.catch(err => {
-			if (handleHttpErrors(req, res, err)) {
-				return;
+export const whoami: RequestHandler = async (req, res) => {
+	try {
+		let user: AnyObject = await getUser(req);
+
+		// If the user email is available, then the record was successfully
+		// populated by the passport middleware
+		if (user.email == null) {
```

Comment:
Keep in mind that in the cloud API we do have users with a `null` email.

Just wanted to clarify that the `most likely` part since I don't see anything wrong on running this extra code anyway, but I would prefer it if there was a different prop that we could check. Is there any chance this would be `undefined` when from a scoped api key instead of `null` when the user genuinely doesn't have an email set?

Reply:
@thgreasi thanks for the heads up. With a scoped access token, the `email` value will always be `undefined`, so we can just check for that